CMakeModules/CompilerSettings.cmake

```diff
@@ -1,14 +1,14 @@
 # GCC
 if(CMAKE_COMPILER_IS_GNUCXX)
-    add_definitions(-std=c++11 -W -Wall -g -Wextra -Wpedantic -Wno-missing-field-initializers -Wno-unused-parameter)
+    add_definitions(-std=c++11 -W -Wall -g -Wextra -Wpedantic -Wno-unused-parameter)
```

Comment:
I believe we could remove `-Wno-unused-parameter` as well. One case could be a problem I can think of is using variables that are only used in assertions. For that case, we could use a macro to suppress the warning as [we did in DART](https://github.com/dartsim/dart/pull/646).